pkg/util/watchdog/watchdog.go

```diff
@@ -24,25 +24,30 @@ func (w *SimpleWatchdog) Start(stop <-chan struct{}) {
 
 	for {
 		ctx, cancel := context.WithCancel(context.Background())
+
+		cleanupCh := make(chan struct{})
 		go func() {
-			<-stop
+			select {
+			case <-stop:
+			case <-cleanupCh:
+			}
 			cancel()
 		}()
+
 		select {
 		case <-ticker.C:
-			select {
-			case <-stop:
-			default:
-				if err := w.onTick(ctx); err != nil {
-					w.OnError(err)
-				}
```

Comment:
actually we already have a channel, we can wait on `ctx.Done()`. So we call `cancel` here at the end of `default` and wait on `ctx.Done()` in the goroutine? Then we have in both cases that the goroutine exits, either via `stop` or via `Done()